cloudstack_plugin/volume.py

```diff
@@ -0,0 +1,105 @@
+########
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# * See the License for the specific language governing permissions and
+# * limitations under the License.
+
+import copy
+from cloudify import ctx
+from cloudify.decorators import operation
+from cloudify.exceptions import NonRecoverableError
+
+from cloudstack_plugin.cloudstack_common import (
+    get_cloud_driver,
+    get_resource_id,
+    CLOUDSTACK_ID_PROPERTY,
+    CLOUDSTACK_NAME_PROPERTY,
+    CLOUDSTACK_TYPE_PROPERTY,
+    COMMON_RUNTIME_PROPERTIES_KEYS
+
+)
+
+VOLUME_CLOUDSTACK_TYPE = 'volume'
+
+RUNTIME_PROPERTIES_KEYS = COMMON_RUNTIME_PROPERTIES_KEYS
+
+
+@operation
+def create(**kwargs):
+    """ Create a volume
+    """
+
+    cloud_driver = get_cloud_driver(ctx)
+
+    volume = {}
+
+    ctx.logger.debug('reading volume attributes.')
+    volume.update(copy.deepcopy(ctx.node.properties['volume']))
+
+    volume_name = volume['name']
+    volume_size = volume['size']
+
```

Comment:
if 'name' or 'size' are not part of the volume, a KeyError will be raised, which will be translated into a recoverable error, which is not what we want. 
check if these keys exist and if not than raise a NonRecoverableError